### PR TITLE
Custom snippet lengths; provision_elastic mgmt command; search type switching

### PR DESCRIPTION
**Pull request by ColeDCrawford**
- Custom snippet lengths
- Use Asyncio for vectorization within Docker; still too slow to be usable
- Data fixtures (seem to crash ECS task due to memory issues, switched to PSQL dump)
- Refactor ES ingest functionality and add `provision_elastic` management command
- Update `example.env`
- Custom management command for `createsuperuser`
- Fix memory crash in ES ingest due to large querySets
- Allow users to switch between search types: `match` (Searchkit default), `simple_query_string` (with operators like `+` and `|`), and `knn`

### CODE REVIEW

**Comment by ColeDCrawford** [comment]
Styling example for the switching: https://codepen.io/Kelderic/pen/Qjagjz

**Review by rcavezza — APPROVED** [review]

File: dle/entrypoint [hunk 1]
```diff
@@ -22,9 +23,21 @@ if [[ ($INIT_SUPERUSER) && ("$INIT_SUPERUSER" = "True") ]]; then
...
+if [[ ($LOAD_FIXTURES) && ("$LOAD_FIXTURES" = "True") ]]; then
+  echo "Loading fixtures"
+  python3 manage.py loaddata /app/data/fixtures/*.json
+  echo "Finished loading fixtures"
+fi
+
+
+# Otherwise, if not loading fixtures, you can run a scrape for new data
+# This will take a long time
 if [[ ($LOAD) && ("$LOAD" = "True") ]]; then
```
Comment (KingYiuWoo) [inline dle/entrypoint:41]: Add HC as well?

File: dle/.env.example [hunk 2]
```diff
@@ -77,18 +77,29 @@ ELASTICSEARCH_USER=elastic
...
-# In the entrypoint, whether to load Django data to Elasticsearch; whether to create an ES NLP pipeline with the BERT model (script doesn't exist yet);
-# and/or whether to load vectors from an external source (don't exist yet);
-# LOAD_DJANGO_TO_ES=True
-# LOAD_VECTORS=False
-# If ES_AUTO_SYNC is True, elasticsearch-django will update the index as data is changed (including on initial ingest)
+# In the entrypoint, whether to load data from vector JSON files into Django
+# Doesn't work well because of complex composite keys / nested structure
+# LOAD_FDA_VECTORS=False
+# LOAD_TGA_VECTORS=False
+# LOAD_EMA_VECTORS=False
```
Comment (KingYiuWoo) [inline dle/.env.example:84]: and HC? Do we still need to keep these load vectors if loading fixtures works?

Comment (ColeDCrawford) [reply dle/.env.example:84]: Loading fixtures sort of works but can also cause containers to crash due to RAM limits ... I think that a PSQL dump is going to be best

File: dle/entrypoint (hunk 1 above)
Comment (ColeDCrawford) [reply dle/entrypoint:41]: Already in `main`, I started this PR before the HC script got merged, so should be good to go

**Pull request merged**